### PR TITLE
feat(mcp): Add pagination parameters to get_cloud_sync_logs tool

### PR DESCRIPTION
**Pull request by aaronsteers**
## Summary
Adds optional parameters to the `get_cloud_sync_logs` MCP tool to prevent sync logs from overwhelming the context window. This addresses an issue where logs were returning ~8.2m tokens and filling up the context.

New parameters:
- `max_lines`: Maximum lines to return (defaults to 4000 if not specified)
- `line_offset`: Skip N lines from the beginning of logs
- `from_tail`: Return lines from the end instead of beginning

Validation ensures `line_offset` and `from_tail` cannot be used together. When logs are truncated, metadata is appended showing how many lines were returned vs total.

## Review & Testing Checklist for Human
- [ ] **Default behavior change**: Previously returned all logs, now defaults to 4000 lines. Verify this default is appropriate for typical use cases.
- [ ] **Edge case**: What happens when `line_offset` exceeds total lines? Currently returns empty result with truncation note - verify this is acceptable UX.
- [ ] **Test manually**: Call `get_cloud_sync_logs` with various parameter combinations (no params, max_lines only, from_tail=True, line_offset=100) against a real connection with logs.

### Notes
- No unit tests were added for the new pagination logic
- The truncation note format `[Showing X of Y total lines]` is appended to the log output

Requested by: AJ Steers (@aaronsteers)
Link to Devin run: https://app.devin.ai/sessions/2b0a9632ae694b73be67301524ed7a8d

## Summary by CodeRabbit

* **New Features**
  * Paginated cloud sync logs: limit lines (defaults to 4000, 0 = no limit), skip from start via offset, or retrieve from end.
  * Logs now return a structured paginated result with text, start line, returned line count, and total available lines.
  * Empty logs return an empty structured result instead of raw text.

* **Bug Fixes**
  * Improved error handling for missing sync jobs/attempts and requested attempts.
  * Validation preventing conflicting pagination options (offset vs. tail).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: airbyte/mcp/cloud_ops.py [hunk 1]
```diff
@@ -825,19 +887,62 @@ def get_cloud_sync_logs(
...
+            job_id=sync_result.job_id,
+            attempt_number=target_attempt.attempt_number,
         )
 
-    return logs
+    # Apply line limiting
+    log_lines = logs.splitlines()
+    total_lines = len(log_lines)
+
+    if max_lines == 0 or total_lines < max_lines:
```
Comment (aaronsteers) [inline airbyte/mcp/cloud_ops.py]: aside - 

```suggestion
    if max_lines == 0 or total_lines <= max_lines:
```
